src/app/test_executive/slot_end_test.ml

```diff
@@ -0,0 +1,210 @@
+open Core
+open Integration_test_lib
+
+module Make (Inputs : Intf.Test.Inputs_intf) = struct
+  open Inputs
+  open Engine
+  open Dsl
+
+  open Test_common.Make (Inputs)
+
+  (* TODO: find a way to avoid this type alias (first class module signatures restrictions make this tricky) *)
+  type network = Network.t
+
+  type node = Network.Node.t
+
+  type dsl = Dsl.t
+
+  let num_extra_keys = 100
+
+  let slot_tx_end = Mina_compile_config.slot_tx_end
```

Comment:
This is conflating the compile-time-config of the test executive with the compile-time-config of the daemon we are running. These are not one in the same, and won't necessarily match.

Reply:
To deal with this, as specified in the comment section of the RFC, these configuration options should be exposed in the node configuration so that the test executive can configure them directly for the test. Then we don't need to pair together various compile time configurations to different test cases.

Reply:
Since overriding these settings in runtime was not added (as per [this discussion](https://github.com/MinaProtocol/mina/pull/14138#discussion_r1370378912)), what would be the best way to do this? I was thinking of adding these parameters to the runtime configuration, but hiding the override compile time config logic behind a feature flag, enabling it only on the build profile used for the integration test (`devnet`?)
Also, Lucy has to be extended to support these new runtime config params, right?

Reply:
Apologies for the confusion from that comment. In the comment you linked, I meant to say that it should not be overridable via the CLI. In another discussion we had mentioned that it should be overridable via the runtime config only. Exposing it via runtime config doesn't have the same risk that exposing it via the CLI does. Most users never edit the runtime config, since it will usually make your node unable to join the chain, and we don't document the runtime config options. https://github.com/MinaProtocol/mina/pull/14138#discussion_r1329357416

Lucy does not require additional changes if you expose this via runtime config. The runtime config is set by every test in Lucy. This option should automatically be exposed, and if it isn't, it's a 2 line change to expose it, since Lucy just takes the test config you give and generates the runtime config from it.

Reply:
Added these two parameters to the runtime configuration with a mechanism to retrieve them if set or default to the compile-time values if not. Also made some small changes to Lucy to use these new config parameters.